openapi_specgen/components.py

```diff
@@ -0,0 +1,58 @@
+
+from typing import List, Union
+from dataclasses import dataclass
+
+@dataclass
+class BasicAuth:
+    protocol_type = "http"
+    scheme = "basic"
+    def as_dict(self):
+        return {
+            'type': self.protocol_type,
+            'scheme': self.scheme
+        }
+
+@dataclass
+class BearerAuth:
+    protocol_type = "http"
+    scheme = "bearer"
+    def as_dict(self):
+        return {
+            'type': self.protocol_type,
+            'scheme': self.scheme
+        }
+
+@dataclass
+class ApiKeyAuth:
+    protocol_type = "apiKey"
+    in_location = "header"
+    name = "X-API-Key"
+    def as_dict(self):
+        return {
+            'type': self.protocol_type,
+            ' in': self.in_location,
+            'name': self.name
+        }
+
+SecuritySchema = Union[BasicAuth, BearerAuth, ApiKeyAuth]
+
+
+class ComponentSet:
```

Comment:
Since this class deals only with Authentication can we rename it to OpenApiAuth or OpenApiSecurity ?

On docstring change to `represent an OpenApi Authentication` and link to this portion of the documentation https://swagger.io/docs/specification/authentication/

Reply:
it contains also `schemas`and builds the whole components branch. I don't like plural name for classes so I append `set`

Reply:
I rename the whole file due to redisign